runtime/lua/vim/lsp/rpc.lua

```diff
@@ -485,14 +457,15 @@ local function start(cmd, cmd_args, dispatchers, extra_spawn_params)
 
   ---@private
   local function handle_body(body)
-    local decoded, err = json_decode(body)
-    if not decoded then
-      -- on_error(client_errors.INVALID_SERVER_JSON, err)
+    local ok, decoded = pcall(vim.json.decode, body)
+    if not ok then
+      on_error(client_errors.INVALID_SERVER_JSON, decoded)
```

Comment:
For reference: this was commented-out in f75be5e9d510d5369c572cf98e78d9480df3b0bb #12655 accidentally.